### ISSUE TITLE
Abouting reproducing ReFT

### ISSUE THREAD

**Issue by DoubleClass** [lqtrung1998/mwp_ReFT#11]
Thank you for your code!

I have some issues with the exact step to reproduce results in Table2 (CodeLLAMA + ReFT).
If I want to follow the method described in your paper exactly the same, I need to :
1. run `bash exps/paper_exps/SFT/gsm8k.sh` , and change the epoch to 2 (the warm up stage) (or maybe I need other more changes in the config file?)
2. get the warmed model and rename it to something like 'ppo_paper_final_new/_models_outputs_sft/gsm8k_nl_codellama/global_step_312_epoch_2'
3. run `bash exps/paper_exps/ReFT/gsm8k.sh` to get the final results

or can I use the hugging face model `Codellama-7b-hf-SFT-warmup-GSM8k` as the starting model, put this model name or path in the `bash exps/paper_exps/ReFT/gsm8k.sh`  'ref_model_name_or_path' and run this script?

Will both these two ways produce the results in the paper? Or I have some misunderstanding? Can you help me with some more detailed instructions? (maybe how to get the warmed up model and start the RL phase)

Thank you very much!

**Comment by lqtrung1998** [comment]
Hi @DoubleClass ,
Thank you for your interest in our work.

Yes, these two ways will reproduce the result in the paper


> 1. run bash exps/paper_exps/SFT/gsm8k.sh , and change the epoch to 2 (the warm up stage) (or maybe I need other more changes in the config file?)

By default, we train the model for 40 epoch and take the epoch 2 checkpoint as initialization for ReFT training, there may be small difference if the SFT training epoch is set to 2 (e.g due to learning rate scheduler....)


Best Regards,
Trung

**Comment by DoubleClass** [comment]
Thank you very much for your help!
